### PR TITLE
Support transforms as AbstractDict

### PR DESCRIPTION
**Pull request by omus**
Somewhat requires DataStreams.jl changes https://github.com/JuliaData/DataStreams.jl/pull/64. Without the DataStreams.jl changes passing anything but a `Dict` will fail.

### CODE REVIEW

File: src/CSV.jl [hunk 1]
```diff
@@ -1,6 +1,7 @@
 __precompile__(true)
 module CSV
 
+using Base: Callable
```
Comment (nalimilan) [inline src/CSV.jl]: Is this related? Otherwise, looks good.

Comment (omus) [reply src/CSV.jl]: I believe I anticipated I would need this and then forgot to remove it. I'll take it out.

**Comment by omus** [comment]
Note that `AbstractDict` was added to Compat.jl in v0.41 which happens to be the version of Compat already required.
